File: GoldenReports.Application/src/DTOs/Reports/ReportDto.cs
﻿using GoldenReports.Application.DTOs.Assets;
using GoldenReports.Application.DTOs.Common;

namespace GoldenReports.Application.DTOs.Reports;

public record ReportDto : EntityDto
{
    public Guid? ParentId { get; init; }
    
    public Guid NamespaceId { get; init; }
    
    public string Name { get; init; } = null!;
    
    public string? Description { get; init; }
    
    public string? Query { get; init; }
    
    public string? Styles { get; init; }
    
    public string? Template { get; init; }

    public IEnumerable<ReportParameterDto> Parameters { get; init; } = new List<ReportParameterDto>();

    public IEnumerable<ReportVariableDto> Variables { get; init; } = new List<ReportVariableDto>();

    public IEnumerable<AssetDto> Assets { get; init; } = new List<AssetDto>();
}

File: GoldenReports.UI.old/old/src/app/core/api/models/report-dto.ts
/* tslint:disable */
/* eslint-disable */
import { AssetDto } from './asset-dto';
import { ReportParameterDto } from './report-parameter-dto';
import { ReportVariableDto } from './report-variable-dto';
export interface ReportDto {
  assets?: Array<AssetDto>;
  creationDate?: string;
  description?: null | string;
  id?: string;
  modificationDate?: string;
  name?: string;
  namespaceId?: string;
  parameters?: Array<ReportParameterDto>;
  parentId?: null | string;
  query?: null | string;
  styles?: null | string;
  template?: null | string;
  variables?: Array<ReportVariableDto>;
}
